### ISSUE TITLE
Assertion failed 'firstILBlock->hasProfileWeight()'

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#10461]
**Repro (of many)**
x64 checked
set COMPlus_JitStress=2
run coreclr tests

**Example CI run**
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitstress2/124/

**Sample output**
```
D:\j\workspace\x64_checked_w---8a0029a8\bin\tests\Windows_NT.x64.Checked\TestWrappers\JIT.Directed\JIT.Directed.XUnitWrapper.cs(21527): error : JIT_Directed._nullabletypes_castclassinterface_r_castclassinterface_r_._nullabletypes_castclassinterface_r_castclassinterface_r_cmd [FAIL] [D:\j\workspace\x64_checked_w---8a0029a8\tests\runtest.proj]
        
        Assert failure(PID 3148 [0x00000c4c], Thread: 3532 [0x0dcc]): Assertion failed 'firstILBlock->hasProfileWeight()' in 'NullableTest5:BoxUnboxToNQ(ref):bool' (IL size 46)
        
            File: d:\j\workspace\x64_checked_w---8a0029a8\src\jit\flowgraph.cpp Line: 12704
            Image: D:\j\workspace\x64_checked_w---8a0029a8\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\CoreRun.exe
```

**Comment by RussKeldorph** [comment]
@dotnet/jit-contrib

**Comment by briansull** [comment]
PR with fix #10466
